packages/eas-cli/src/update/queries.ts

```diff
@@ -2,84 +2,180 @@ import chalk from 'chalk';
+}: {
+  projectId: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<void> {
+  if (paginatedQueryOptions.nonInteractive) {
+    const updateGroups = await queryUpdateGroupsOnAppAsync({
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
+      appId: projectId,
+    });
+    renderUpdateGroupsAsTableWithBranchColumn(updateGroups, paginatedQueryOptions);
   } else {
     await paginatedQueryWithConfirmPromptAsync({
-      limit: options.limit ?? UPDATES_LIMIT,
-      offset: options.offset,
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
       queryToPerform: (limit, offset) =>
-        queryUpdateGroupsForBranchAsync(limit, offset, projectId, branchName),
+        queryUpdateGroupsOnAppAsync({ limit, offset, appId: projectId }),
       promptOptions: {
-        title: 'Load more branches?',
-        renderListItems: updates => renderUpdateGroups(updates, options, branchName),
+        title: 'Load more update groups?',
+        renderListItems: updateGroup =>
+          renderUpdateGroupsAsTableWithBranchColumn(updateGroup, paginatedQueryOptions),
       },
     });
   }
 }
 
-async function queryUpdateGroupsForBranchAsync(
-  limit: number,
-  offset: number,
-  projectId: string,
-  branchName: string
-): Promise<BranchUpdateObject[]> {
-  const { app } = await UpdateQuery.viewBranchAsync({
-    appId: projectId,
-    name: branchName,
+export async function listAndRenderUpdateGroupsOnBranchAsync({
+  projectId,
+  branchName,
+  paginatedQueryOptions,
+}: {
+  projectId: string;
+  branchName: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<void> {
+  if (paginatedQueryOptions.nonInteractive) {
+    const updateGroups = await queryUpdateGroupsOnBranchAsync({
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
+      appId: projectId,
+      branchName,
+    });
+    renderUpdateGroupsAsTable(updateGroups, paginatedQueryOptions);
+  } else {
+    await paginatedQueryWithConfirmPromptAsync({
+      limit: paginatedQueryOptions.limit ?? UPDATE_GROUPS_LIMIT,
+      offset: paginatedQueryOptions.offset,
+      queryToPerform: (limit, offset) =>
+        queryUpdateGroupsOnBranchAsync({ limit, offset, appId: projectId, branchName }),
+      promptOptions: {
+        title: 'Load more update groups?',
+        renderListItems: updates => renderUpdateGroupsAsTable(updates, paginatedQueryOptions),
+      },
+    });
+  }
+}
+
+export async function selectUpdateGroupOnBranchAsync({
+  projectId,
+  branchName,
+  paginatedQueryOptions: painatedQueryOptions,
+}: {
+  projectId: string;
+  branchName: string;
+  paginatedQueryOptions: PaginatedQueryOptions;
+}): Promise<BranchUpdateGroupObject> {
+  if (painatedQueryOptions.nonInteractive) {
+    throw new Error('Unable to select an update in non-interactive mode.');
+  } else {
```

Comment:
no need for the else (can just move all the code in the else out of the clause since the if throws)